examples/copilot/open_ai_copilot.py

```diff
@@ -0,0 +1,252 @@
+                                    ],
+                                    max_tokens=2048)
+
+    function_call = DynamicTask(task_reference_name='fn_call_ref', dynamic_task='SUB_WORKFLOW')
+    function_call.input_parameters['steps'] = chat_complete.output('function_parameters.steps')
+    function_call.input_parameters['inputs'] = chat_complete.output('function_parameters.inputs')
+    function_call.input_parameters['subWorkflowName'] = 'copilot_execution'
+    function_call.input_parameters['subWorkflowVersion'] = 1
+
+    sub_workflow = SubWorkflowTask(task_ref_name='execute_workflow', workflow_name='copilot_execution', version=1)
+
+    create = create_workflow(task_ref_name='create_workflow', steps=chat_complete.output('function_parameters.steps'),
+                             inputs=chat_complete.output('function_parameters.inputs'))
+    call_function = SwitchTask(task_ref_name='to_call_or_not', case_expression=chat_complete.output('function'))
+    call_function.switch_case('create_workflow', [create, sub_workflow])
+
+    call_one_fun = DynamicTask(task_reference_name='call_one_fun_ref', dynamic_task=chat_complete.output('function'))
+    call_one_fun.input_parameters['inputs'] = chat_complete.output('function_parameters')
+    call_one_fun.input_parameters['dynamicTaskInputParam'] = 'inputs'
+
+    call_function.default_case([call_one_fun])
+
+    wf >> user_input >> chat_complete >> call_function
+
+    # let's make sure we don't run it for more than 2 minutes -- avoid runaway loops
+    wf.timeout_seconds(120).timeout_policy(timeout_policy=TimeoutPolicy.TIME_OUT_WORKFLOW)
+    message = """
+    I am a helpful bot that can help with your customer management. 
+    
+    Here are some examples:
+    
+    1. Get me the list of top N customers
+    2. Get the list of all the customers
+    3. Get the list of top N customers and send them a promo code
+    """
+    print(message)
+    workflow_run = wf.execute(wait_until_task_ref=user_input.task_reference_name, wait_for_seconds=120)
+    workflow_id = workflow_run.workflow_id
+    query = input('>> ')
+    input_task = workflow_run.get_task(task_reference_name=user_input.task_reference_name)
+    workflow_run = workflow_client.update_state(workflow_id=workflow_id,
+                                                update_requesst=WorkflowStateUpdate(
+                                                    task_reference_name=user_input.task_reference_name,
+                                                    task_result=TaskResult(task_id=input_task.task_id, output_data={
+                                                        'query': query
+                                                    }, status=TaskResultStatus.COMPLETED)
+                                                ),
+                                                wait_for_seconds=30)
+
+    print(f'https://pg-qa.orkesconductor.com/execution/{workflow_id}')
```

Comment:
This url should comes from the environment variable.

Reply:
done.